### DISCUSSION TITLE
aws sso check makes terminal start slow.

### DISCUSSION THREAD

**Discussion by LilMonk** [aws/aws-cli#8509] [Q&A]
Whenever I open up a new terminal window the aws cli checks the validity of the access tokens. This process delay the terminal start. Is there any way to lazily check the sso token expiry when needed.

**Comment by uldyssian-sh** [comment]
AWS CLI does not run token checks automatically on terminal startup by itself. The slowdown is caused by your shell configuration (prompt/rc scripts) running an AWS command (often `aws sts get-caller-identity` or a custom “aws sso check”) every time a new shell starts or every time the prompt is rendered.

To make it lazy, you need to stop running AWS CLI at startup and instead read local cache state (fast) or only call AWS when you actually run an AWS command.

Fast, local “is SSO token still valid?” check:

AWS SSO tokens are cached as JSON files under:

~/.aws/sso/cache/

Those files contain an `expiresAt` timestamp. Reading that timestamp is just a local file read and is effectively instant, unlike calling STS.

So, for your prompt/startup logic:
- do NOT call `aws sts ...` or `aws sso ...` on shell init
- instead read `expiresAt` from ~/.aws/sso/cache and only trigger `aws sso login` when you actually need AWS access

Also, if you currently run a login check on every prompt render (PS1), change it to run at most once per N seconds (cache the result in a temp file) to avoid repeated AWS calls.

Summary:

The fix is not in AWS CLI; it’s in your shell startup/prompt scripts.
Remove AWS CLI calls from terminal initialization, and use the local SSO cache (expiresAt) or a throttled check so the terminal opens instantly and authentication is checked only when needed.

**Reply by uldyssian-sh** [reply]
Thanks! If this helps, please consider marking it as “Helpful” or “Accepted” 😊
